rt/rs/extensions/rx2/src/main/java/org/apache/cxf/jaxrs/rx2/client/FlowableRxInvokerImpl.java

```diff
@@ -147,34 +150,90 @@ public FlowableRxInvokerImpl(WebClient wc, ExecutorService ex) {
 
     @Override
     public <T> Flowable<T> method(String name, Entity<?> entity, Class<T> responseType) {
+        
+        Flowable<T> flowable = Flowable.create(new FlowableOnSubscribe<T>() {
+            @Override
+            public void subscribe(FlowableEmitter<T> emitter) throws Exception {
+                    try {
+                        T response = syncInvoker.method(name, entity, responseType);
+                        emitter.onNext(response);
+                        emitter.onComplete();
+                    } catch (Throwable e) {
+                        emitter.onError(e);
+                    }
+            }
+        }, BackpressureStrategy.DROP);
+        
         if (sc == null) {
-            return Flowable.fromFuture(wc.async().method(name, entity, responseType));
+            return flowable;
         }
-        return Flowable.fromFuture(wc.async().method(name, entity, responseType), sc);
+        return flowable.subscribeOn(sc).observeOn(sc);        
     }
 
     @Override
     public <T> Flowable<T> method(String name, Entity<?> entity, GenericType<T> responseType) {
+        
+        Flowable<T> flowable = Flowable.create(new FlowableOnSubscribe<T>() {
+            @Override
+            public void subscribe(FlowableEmitter<T> emitter) throws Exception {
```

Comment:
We may generalize code a bit, too many repetition I think:
```
    private <T> Flowable<T> create(Supplier<T> supplier) {
        Flowable<T> flowable = Flowable.create(new FlowableOnSubscribe<T>() {
            @Override
            public void subscribe(FlowableEmitter<T> emitter) throws Exception {
                try {
                    T response = supplier.get();
                    if (!emitter.isCancelled()) {
                        emitter.onNext(response);
                    }
                    
                    if (!emitter.isCancelled()) {
                        emitter.onComplete();
                    }
                } catch (Throwable e) {
                    if (!emitter.isCancelled()) {
                        emitter.onError(e);
                    }
                }
            }
        }, BackpressureStrategy.DROP);
        
        if (sc == null) {
            return flowable;
        }
        
        return flowable.subscribeOn(sc).observeOn(sc);
    }
```

Then the implementation is just needs a supplier:
```
   @Override
    public <T> Flowable<T> method(String name, Entity<?> entity, GenericType<T> responseType) {
        return create(() -> syncInvoker.method(name, entity, responseType));
    }

    @Override
    public <T> Flowable<T> method(String name, Class<T> responseType) {
        return create(() -> syncInvoker.method(name, responseType));
    }

    @Override
    public <T> Flowable<T> method(String name, GenericType<T> responseType) {
        return create(() -> syncInvoker.method(name, responseType));
    }
```

What do you think?

Reply:
Agree.  Thanks for the comments.  Done.